backend/search.js

```diff
@@ -385,40 +392,57 @@ class Search {
 
     // Cache the refs.
     let refs;
+    let subCount;
+    let subName;
     if (cacheEntry) {
       refs = this.refCache[termId + searchTerm].refs;
       wasSubjectMatch = this.refCache[termId + searchTerm].wasSubjectMatch;
+      // if (wasSubjectMatch) {
+      //     const possibleSubjectMatch = this.checkForSubjectMatch(searchTerm, termId);
+      //     subCount = possibleSubjectMatch.subjectCount;
+      //     subName = possibleSubjectMatch.subjectName;
+
+      // }
 
       // Update the timestamp of this cache item.
       this.refCache[termId + searchTerm].time = Date.now();
     } else {
+      // if subject matches, make sure to get appropriate info needed
       const possibleSubjectMatch = this.checkForSubjectMatch(searchTerm, termId);
       if (possibleSubjectMatch) {
-        refs = possibleSubjectMatch;
+        refs = possibleSubjectMatch.results;
+        subCount = possibleSubjectMatch.subjectCount;
+        subName = possibleSubjectMatch.subjectName;
         wasSubjectMatch = true;
       } else {
         refs = this.getRefs(searchTerm, termId);
       }
 
       this.itemsInCache++;
 
+      // if the subject is null, then the extra code won't be triggered
       this.refCache[termId + searchTerm] = {
         refs: refs,
         wasSubjectMatch: wasSubjectMatch,
+        subjectName: subName,
+        subjectCount: subCount,
         time: Date.now(),
       };
     }
 
     const analytics = {
       status: 'Success',
       wasSubjectMatch: wasSubjectMatch,
+      subjectCount: subCount,
```

Comment:
would it be possible to change up the return value of this function a bit? SO like instead of returning an analytics object and a results(`[...]`) object - return something like this

```js
{
  resultsObject: {
    results: [...]
   subjectName: ...
   subjectCount:...
  wasSubjectMatch: ...
 } 
analytics: ... 
```

I think if we did this we could simplify some things in backend/server.js too - check the comment there

Reply:
ok yeah cause if we change it then it'll always have the thingymadoos